packages/browser/src/BacktraceBrowserSessionProvider.ts

```diff
@@ -0,0 +1,72 @@
+import { BacktraceSessionProvider, IdGenerator } from '@backtrace/sdk-core';
+import { TimeHelper } from '@backtrace/sdk-core/lib/common/TimeHelper';
+
+export class BacktraceBrowserSessionProvider implements BacktraceSessionProvider {
+    /**
+     * Session persistence interval. If no event was send in the persistence interval
+     * the session is treaten as an old session.
+     */
+    private readonly PERSISTENCE_INTERVAL = TimeHelper.convertSecondsToMilliseconds(30 * 60);
+    private readonly SESSION_LAST_ACTIVE = 'backtrace-last-active';
+    private readonly SESSION_GUID = 'backtrace-guid';
+
+    /**
+     * Session last active timestamp
+     */
+    get lastActive(): number {
+        return this._lastActive;
+    }
+    /**
+     * Determinates if the current session is new.
+     */
+    public readonly newSession: boolean = true;
+    /**
+     * Current session Id
+     */
+    public readonly sessionId: string = IdGenerator.uuid();
+
+    private _lastActive = 0;
+
+    constructor() {
+        if (!window.localStorage) {
+            return;
+        }
+
+        const lastActive = this.readLastActiveTimestamp();
+        if (!lastActive || TimeHelper.now() - lastActive > this.PERSISTENCE_INTERVAL) {
+            this.updateLastActiveTimestamp();
+            localStorage.setItem(this.SESSION_GUID, this.sessionId);
+            return;
+        }
```

Comment:
This information needs to exist asap in the storage, otherwise, when someone opens another tab quickly, the application will have two sessions. This is why I decided to add it to the constructor. I agree it might be weird, however, this is a real situation. 

We can move it to the store method if you would prefer, but the store method will be invoked just after the session provider creation.

I will move it to the start method that we start to use once after client initialization